lonestar/analytics/cpu/connected-components/connected_components_cli.cpp

```diff
@@ -0,0 +1,244 @@
+    plan = ConnectedComponentsPlan::Synchronous();
+    break;
+  case ConnectedComponentsPlan::kAsynchronous:
+    plan = ConnectedComponentsPlan::Asynchronous();
+    break;
+  case ConnectedComponentsPlan::kEdgeAsynchronous:
+    plan = ConnectedComponentsPlan::EdgeAsynchronous();
+    break;
+  case ConnectedComponentsPlan::kEdgeTiledAsynchronous:
+    galois::gInfo("INFO: Using edge tile size: ", edgeTileSize);
+    galois::gInfo("WARNING: Performance may vary due to parameter");
+    plan = ConnectedComponentsPlan::EdgeTiledAsynchronous(edgeTileSize);
+    break;
+  case ConnectedComponentsPlan::kBlockedAsynchronous:
+    plan = ConnectedComponentsPlan::BlockedAsynchronous();
+    break;
+  case ConnectedComponentsPlan::kAfforest:
+    galois::gInfo(
+        "INFO: Using neighbor sample size: ", neighborSampleSize,
+        " component sample frequency: ", componentSampleFrequency);
+    galois::gInfo("WARNING: Performance may vary due to the parameters");
+    plan = ConnectedComponentsPlan::Afforest(
+        neighborSampleSize, componentSampleFrequency);
+    break;
+  case ConnectedComponentsPlan::kEdgeAfforest:
+    galois::gInfo(
+        "INFO: Using neighbor sample size: ", neighborSampleSize,
+        " component sample frequency: ", componentSampleFrequency);
+    galois::gInfo("WARNING: Performance may vary due to the parameters");
+    plan = ConnectedComponentsPlan::EdgeAfforest(
+        neighborSampleSize, componentSampleFrequency);
+    break;
+  case ConnectedComponentsPlan::kEdgeTiledAfforest:
+    galois::gInfo(
+        "INFO: Using edge tile size: ", edgeTileSize,
+        " neighbor sample size: ", neighborSampleSize,
+        " component sample frequency: ", componentSampleFrequency);
+    galois::gInfo("WARNING: Performance may vary due to the parameters");
+    plan = ConnectedComponentsPlan::EdgeTiledAfforest(
+        neighborSampleSize, componentSampleFrequency);
+    break;
+  case ConnectedComponentsPlan::kAutomatic:
+    galois::gInfo(
+        "INFO: Using neighbor sample size: ", neighborSampleSize,
+        " component sample frequency: ", componentSampleFrequency);
+    galois::gInfo("WARNING: Performance may vary due to the parameters");
+    plan = ConnectedComponentsPlan::Automatic(
+        neighborSampleSize, componentSampleFrequency);
+    break;
+  default:
+    std::cerr << "Invalid algorithm\n";
+    abort();
+  }
+
+  auto pg_result = ConnectedComponents(pfg.get(), "component", plan);
+  if (!pg_result) {
+    GALOIS_LOG_FATAL(
+        "Failed to run ConnectedComponents: {}", pg_result.error());
+  }
+
+  auto stats_result =
+      ConnectedComponentsStatistics::Compute(pfg.get(), "component");
+  if (!stats_result) {
+    GALOIS_LOG_FATAL(
+        "Failed to compute ConnectedComponents statistics: {}",
+        stats_result.error());
+  }
+  auto stats = stats_result.value();
+  stats.Print();
+
+  if (!skipVerify) {
+    if (!ConnectedComponentsValidate(pfg.get(), "component")) {
```

Comment:
I wrote this initially too, but it's incorrect. The correct version is `auto r = ConnectedComponentsValidate(pfg.get(), "component"); r && r.value()` otherwise you will conflate successful check with a validated graph. 
https://github.com/KatanaGraph/katana/blob/master/lonestar/analytics/cpu/bfs/bfs_cli.cpp#L136

Reply:
oops.